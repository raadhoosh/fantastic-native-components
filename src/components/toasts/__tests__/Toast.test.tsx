import Toast from "../index";
import { shallow, ShallowWrapper } from "enzyme";
import React from "react";

const createTestProps = (props: Object) => ({
  ...props,
});

describe("Toast", () => {
  describe("rendering", () => {
    let component: ShallowWrapper;
    let props: Object;
    beforeEach(() => {
      props = createTestProps({
        color: '#f00',
        primary: true,
        secondary: true,
        success: true,
      });
      component = shallow(<Toast primary 
        color= '#f00'         
        backgroundColor="9c27b0"        
      />);
    });
    it("renders", () => {
      expect(component.exists()).toBe(true);
  });
  it("toMatchSnapshot", () => {
      expect(component).toMatchSnapshot();
  });
  });
});
