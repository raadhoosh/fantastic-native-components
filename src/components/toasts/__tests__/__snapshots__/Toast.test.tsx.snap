// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast rendering toMatchSnapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Toasts
    backgroundColor="9c27b0"
    color="#f00"
    primary={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <ForwardRef
        backgroundColor="9c27b0"
        color="#f00"
        primary={true}
        style={
          Object {
            "transform": Array [
              Object {
                "translateY": -70,
              },
            ],
          }
        }
      >
        <ForwardRef
          backgroundColor="9c27b0"
          color="#f00"
          primary={true}
        />
      </ForwardRef>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "backgroundColor": "9c27b0",
        "children": <ForwardRef
          backgroundColor="9c27b0"
          color="#f00"
          primary={true}
        />,
        "color": "#f00",
        "primary": true,
        "style": Object {
          "transform": Array [
            Object {
              "translateY": -70,
            },
          ],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "backgroundColor": "9c27b0",
          "children": undefined,
          "color": "#f00",
          "primary": true,
        },
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "rules": Array [
              "       
    color: ",
              [Function],
              "; 
    font-size:",
              [Function],
              "; 
",
            ],
          },
          "displayName": "Styled(Component)",
          "propTypes": Object {
            "accessible": [Function],
            "adjustsFontSizeToFit": [Function],
            "allowFontScaling": [Function],
            "disabled": [Function],
            "ellipsizeMode": [Function],
            "minimumFontScale": [Function],
            "nativeID": [Function],
            "numberOfLines": [Function],
            "onLayout": [Function],
            "onLongPress": [Function],
            "onPress": [Function],
            "pressRetentionOffset": [Function],
            "selectable": [Function],
            "selectionColor": [Function],
            "style": [Function],
            "suppressHighlighting": [Function],
            "testID": [Function],
            "textBreakStrategy": [Function],
          },
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": [Function],
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "__skipSetNativeProps_FOR_TESTS_ONLY": true,
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "rules": Array [
            "      
    border: 1px solid ",
            [Function],
            ";       
    width: ",
            [Function],
            ";       
    height: ",
            [Function],
            ";             
    border-radius: ",
            [Function],
            ";   
    background-color: ",
            [Function],
            ";      
    position: absolute;
    left: 0;
    right: 0;
    top: -60;  
    padding: 10px;    
    flex:1;
    justify-content: center;
",
          ],
        },
        "displayName": "Styled(AnimatedComponent)",
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": [Function],
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
        Symbol(Symbol.hasInstance): [Function],
      },
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <ForwardRef
          backgroundColor="9c27b0"
          color="#f00"
          primary={true}
          style={
            Object {
              "transform": Array [
                Object {
                  "translateY": -70,
                },
              ],
            }
          }
        >
          <ForwardRef
            backgroundColor="9c27b0"
            color="#f00"
            primary={true}
          />
        </ForwardRef>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "backgroundColor": "9c27b0",
          "children": <ForwardRef
            backgroundColor="9c27b0"
            color="#f00"
            primary={true}
          />,
          "color": "#f00",
          "primary": true,
          "style": Object {
            "transform": Array [
              Object {
                "translateY": -70,
              },
            ],
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "backgroundColor": "9c27b0",
            "children": undefined,
            "color": "#f00",
            "primary": true,
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": false,
              "rules": Array [
                "       
    color: ",
                [Function],
                "; 
    font-size:",
                [Function],
                "; 
",
              ],
            },
            "displayName": "Styled(Component)",
            "propTypes": Object {
              "accessible": [Function],
              "adjustsFontSizeToFit": [Function],
              "allowFontScaling": [Function],
              "disabled": [Function],
              "ellipsizeMode": [Function],
              "minimumFontScale": [Function],
              "nativeID": [Function],
              "numberOfLines": [Function],
              "onLayout": [Function],
              "onLongPress": [Function],
              "onPress": [Function],
              "pressRetentionOffset": [Function],
              "selectable": [Function],
              "selectionColor": [Function],
              "style": [Function],
              "suppressHighlighting": [Function],
              "testID": [Function],
              "textBreakStrategy": [Function],
            },
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": [Function],
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "__skipSetNativeProps_FOR_TESTS_ONLY": true,
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "rules": Array [
              "      
    border: 1px solid ",
              [Function],
              ";       
    width: ",
              [Function],
              ";       
    height: ",
              [Function],
              ";             
    border-radius: ",
              [Function],
              ";   
    background-color: ",
              [Function],
              ";      
    position: absolute;
    left: 0;
    right: 0;
    top: -60;  
    padding: 10px;    
    flex:1;
    justify-content: center;
",
            ],
          },
          "displayName": "Styled(AnimatedComponent)",
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": [Function],
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
